test/globalize_test.rb

```diff
@@ -217,4 +217,28 @@ class GlobalizeTest < MiniTest::Spec
       end
     end
   end
+
+  describe 'with locale' do
+    def perform_with_locale(locale)
+      Thread.new do
+        Globalize.with_locale(locale) do
+          sleep 1
```

Comment:
@simi Could you clarify why is `sleep` needed here?

Is it to give a scheduler time to switch into another thread?
Maybe we could use `Thread.pass` instead